### DISCUSSION TITLE
Question and Suggestion

### DISCUSSION THREAD

**Discussion by JoshuaHintze** [svecosystem/runed#88] [General]
### Change Type

Addition

### Proposed Changes

I came to the github link directly from a post in the Svelte Discord. I tried looking at the readme but I couldn't find the documentation site. Finally I found runed.dev in some sample code. It would be great to add a link to that in the readme.md

Quick question, why is the naming of the runed utilities all over the place.

In some cases you use `useXYZ` in some cases you use PascalCasing like MediaQuery or ElementRect or some cases its camelCase isMounted()

Just curious the reasons.

**Comment by huntabyte** [comment]
> Quick question, why is the naming of the runed utilities all over the place.
> 
> In some cases you use useXYZ in some cases you use PascalCasing like MediaQuery or ElementRect or some cases its camelCase isMounted()

Trust me, we exhausted bikeshedding the naming convention for the various utilities. It's not 100% set in stone, but here's the _general_ idea behind it and how to think of the different casings:

- **PascalCase** - reactive classes and must be instantiated with `new Thing()`
- **camelCase** (activeElement) - reactive global objects that can be imported and whose value can be accessed with `.current`, no need to instantiate
- **useXYZ** - usually wrap functions or `use` is prefixed to avoid conflicting with a DOM API (IntersectionObserver, ResizeObserver), etc.

These aren't hard-set and we're approaching each one on a case-by-case basis.

I'll also see if @TGlide wants to provide any additional mental models/context here to help justify.

**Reply by TGlide** [reply]
This is exactly it. In general, we try to go for classes first. In some rare cases, a function or global object makes more sense, so we change it.

We think it's not that important to have a global naming convention, but rather, appeal to the strength of each approach and use the appropriate naming.

**Comment by JoshuaHintze** [comment]
Sounds good. I was just trying to mentally picture it so I knew what how to start typing so that auto complete will pick it up better. Thanks for the library! If I ever think of some other useful utilities I'll try to contribute.
